lib/omnibus/config.rb

```diff
@@ -333,6 +333,13 @@ def reset!
     # @return [Integer]
     default(:publish_retries, 2)
 
+    # Directory pattern for publisher. Interpolation of matadata keys are supported.
+    #
+    # @example 'custom_prefix/%{platform}/%{platform_version}/%{arch}/%{basename}'
+    #
+    # @return [Integer]
+    default(:publish_dir_pattern, nil)
```

Comment:
Yes, that sounds reasonable. Especially if we want to use both of publishers for the same Omnibus project. 👍 
I've updated the PR, so now we have 2 different options: `artifactory_publish_pattern` and `s3_publish_pattern`, with defaults defined in `Omnibus::Config` class